core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -2140,6 +2140,12 @@ private[zio] trait ZIOFunctions extends Serializable {
   final def fromFunctionM[R, E, A](f: R => IO[E, A]): ZIO[R, E, A] =
     environment[R].flatMap(f)
 
+  /**
+   * Lifts a function returning Future into an effect that requires the input to the function.
+   */
+  final def fromFunctionF[R, A](f: R => scala.concurrent.Future[A]): RIO[R, A] =
```

Comment:
`fromFunctionFuture`?

Reply:
I thought about it originally. Seemed too long, but if you thought about the same then maybe it's actually a good name ;)